### PR TITLE
Add SpaceTime Raster tests

### PR DESCRIPTION
**Pull request by moradology**

### CODE REVIEW

File: spark/src/test/scala/geotrellis/spark/io/hadoop/HadoopSpaceTimeRasterCatalogSpec.scala [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  describe("HadoopRasterCatalog with SpaceTimeKey Rasters") {
+    ifCanRunSpark {
+      val catalogPath = new Path(inputHome, ("st-catalog-spec"))
+      val fs = catalogPath.getFileSystem(sc.hadoopConfiguration)
+      HdfsUtils.deletePath(catalogPath, sc.hadoopConfiguration)
+      val catalog = HadoopRasterCatalog(catalogPath)
+
+      it("ZCurveKeyIndexMethod.byYear") {
+        val coordST = CoordinateSpaceTime
+        catalog.writer[SpaceTimeKey](ZCurveKeyIndexMethod.byYear).write(LayerId("coordST", 10), coordST)
```
Comment (lossyrob) [inline spark/src/test/scala/geotrellis/spark/io/hadoop/HadoopSpaceTimeRasterCatalogSpec.scala]: Formatting is long. 

```scala
catalog.writer[SpaceTimeKey](ZCurveKeyIndexMethod.byYear)
            .write(LayerId("coordST", 10), coordST)
```
would look nicer.

File: spark/src/test/scala/geotrellis/spark/io/hadoop/HadoopSpaceTimeRasterCatalogSpec.scala [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import geotrellis.spark.tiling._
+import geotrellis.raster.op.local._
+import geotrellis.proj4.LatLng
+import geotrellis.spark.testfiles._
+
+import org.scalatest._
+import org.apache.hadoop.fs.Path
+import com.github.nscala_time.time.Imports._
+
+class HadoopSpaceTimeRasterCatalogSpec extends FunSpec
```
Comment (lossyrob) [inline spark/src/test/scala/geotrellis/spark/io/hadoop/HadoopSpaceTimeRasterCatalogSpec.scala]: Usually we have Spec class names line up with actual class names. So these would maybe belong in a `describe` block in `HadoopRasterCatalogSpec`.

**Comment by lossyrob** [comment]
@moradology can you update this PR so I can merge it in

**Comment by moradology** [comment]
I'll jump on this ASAP. slight HL emergency.

**Comment by lossyrob** [comment]
Merging even though travis failed, Travis seems to be having issues and reporting bogus errors.

**Pull request merged**